File: frontend/lib/types/training.ts
export interface FaceImage {
  id: number;
  employee_id: number;
  file_path: string;
  file_hash: string | null;
  width: number | null;
  height: number | null;
  uploaded_by: number | null;
  created_at: string;
}

export interface FaceEmbedding {
  id: number;
  employee_id: number;
  image_id: number;
  dim: number;
  model_name: string;
  quality_score: number;
  created_at: string;
}

export interface TrainingResult {
  employee_id: number;
  accepted: number;
  rejected: number;
  total_embeddings: number;
  errors: string[];
}

File: backend/app/schemas/training.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel

from app.schemas.common import ORMModel


class FaceImageRead(ORMModel):
    id: int
    employee_id: int
    file_path: str
    file_hash: str | None
    width: int | None
    height: int | None
    uploaded_by: int | None
    created_at: datetime


class EmbeddingRead(ORMModel):
    id: int
    employee_id: int
    image_id: int
    dim: int
    model_name: str
    quality_score: float
    created_at: datetime


class TrainingResult(BaseModel):
    employee_id: int
    accepted: int
    rejected: int
    total_embeddings: int
    errors: list[str]
